Edit information front end

byroncjy opened the pull request:
This adds an Edit Information screen to the app (issue 12).
This is built on top of AngelaSprint1, with eslint applied. In doing so, this populates the front-end folder with basic React code.
Note that App.test.js is commented out since its eslint errors cannot be autofixed.

(Before reviewing changes, wait until 10/21/2023 for Mockaroo API call limits to reset)

Resolves #12

SivanC reviewed (changes requested):
Looks really solid! I know some of this is from Angela's Header/Footer/Home but since this will probably be the first merged PR I just made some comments on some of it. Code aside I'm not really familiar with web dev practices but I thought we maybe should write `button` instead of `btn` to keep consistent with no abbreviations in CSS/HTML class names.

As for the logic, there are some things that should probably be changed:
- Don't allow the user to save when there is an empty info input box (or delete it when they save?)
- Should the user be allowed to create more boxes than there are options? If not, then options should also disappear from the dropdown menu as you include them. Probably we will want people to be able to use multiple, though.
- Allow the user to edit their name? Where will we get this information originally, on signup?
- Put the "Add Another Platform" button on top so it doesn't move each time

byroncjy commented on front-end/src/EditInformation.js:
```diff
@@ -0,0 +1,195 @@
+
+      if (response.status === 200) {
+        // Update the profile picture in the UI
+        setProfileData({ ...profileData, url_picture: URL.createObjectURL(image) })
```
Not sure what you mean, in this case when the user uploads a new profile pic, we send a PUT request to update the image in the database. This procedure is still subject to change though, since we don't yet know how we're going to receive images through the API.

XinranTaoAngela replied:
Looks good, just one question we can discuss as a group is when we select what credential to be add(the drop down menu), if the user want to add anything that is not shown on our drop down list, should we add an "other" option to let them manually add the name of the platform and the link to it? This is just a question to be considered.

byroncjy commented on front-end/src/EditInformation.js:
```diff
@@ -0,0 +1,195 @@
+      )
+
+      if (response.status !== 200) {
+        throw new Error('Network response was not ok')
```
Done

byroncjy commented on front-end/src/EditInformation.js:
```diff
@@ -0,0 +1,195 @@
+        {/* Add new entry */}
+        <button onClick={handleAddPlatformInformation}>Add another platform</button>
+        {/* Save updated platform information */}
+        <button type="button" className="save-btn" onClick={handleSave}>
```
Done

byroncjy commented on front-end/src/EditInformation.js:
```diff
@@ -0,0 +1,195 @@
+  const handleSave = async () => {
+    try {
+      const requestData = platformInformationMap.map((item) => {
+        return {
```
Map was needed because of different key names in API vs internal map. This is now standardized, so map removed. 
Also updated to not save "Select Platform" or empty info entries.

byroncjy commented on front-end/src/HomeScreen.js:
```diff
@@ -0,0 +1,21 @@
+        <div className="home-links">
+          <a href="/edit-information" className="home-link">Edit Information</a>
+          <a href="/saved-connections" className="home-link">Saved Connections</a>
+          <a href="/generate-code" className="home-link">Generate Code</a>
```
Agreed, updated

byroncjy commented on front-end/src/App.js:
```diff
@@ -0,0 +1,30 @@
+        <Route path="/home" element={<HomeScreen />} />
+        <Route path="/edit-information" element={<EditInformation />} />
+        <Route path="/saved-connections" element={<SavedConnections />} />
+        <Route path="/generate-code" element={<GenerateCode />} />
```
Updated. We will also need another route to the actual "My QR Code" screen which we can update once finalized

byroncjy commented on front-end/src/Header.css:
```diff
@@ -0,0 +1,30 @@
+    text-align: center;
+}
+
+.back-btn, .logout-btn {
```
Updated

byroncjy commented:
How I addressed the issues: 

Don't allow the user to save when there is an empty info input box (or delete it when they save?)
- Updated to only save non-empty platform and info entries (empty entries aren't saved)

Should the user be allowed to create more boxes than there are options? If not, then options should also disappear from the dropdown menu as you include them. Probably we will want people to be able to use multiple, though.
- For now, updated such that duplicate platforms aren't allowed since it might cause issues with storing to database.

Allow the user to edit their name? Where will we get this information originally, on signup?
- Signup should prompt for names initially but not sure. Regardless, updated to allow editing of names.

Put the "Add Another Platform" button on top so it doesn't move each time
- No change, I think it's more intuitive to have "add another platform" in the position of where the new platform would get added even though it'll get moved down

SivanC commented on front-end/src/EditInformation.js:
```diff
@@ -102,19 +104,61 @@ const EditInformation = () => {
+    try {
+      // Check for duplicate entries
+      const platformNames = new Set()
+      let hasDuplicates = false
```
Don't think the flag is necessary since elements are unique in a set, can just check the length after adding everything,

byroncjy replied:
True lol, will address it next PR

SivanC reviewed (approved):
Looking great! Changes all look good. Only minor thing is that for checking platforms you should probably either use a set or a flag, not both, but not super urgent/important. Go ahead and merge if you're feeling good about it.
